plugins/redfish/fu-redfish-multipart-device.c

```diff
@@ -108,22 +126,24 @@ fu_redfish_multipart_device_write_firmware(FuDevice *device,
 	}
 
 	/* prefer the header, otherwise fall back to the response */
-	json_obj = fu_redfish_request_get_json_object(request);
-	if (json_object_has_member(json_obj, "TaskMonitor")) {
-		const gchar *tmp = json_object_get_string_member(json_obj, "TaskMonitor");
-		g_debug("task manager for cleanup is %s", tmp);
-	}
-
-	/* poll the task for progress */
-	if (!json_object_has_member(json_obj, "@odata.id")) {
-		g_set_error(error,
-			    FWUPD_ERROR,
-			    FWUPD_ERROR_INVALID_FILE,
-			    "no task returned for %s",
-			    fu_redfish_backend_get_push_uri_path(backend));
-		return FALSE;
+	if (g_utf8_strlen(location, 1) == 0) {
```

Comment:
I think we need to check for location==NULL and if so set a GError and return False.